### ISSUE TITLE
Add Social Activity Feed to My Books page

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#10242]
## Context

* See: #7704; the `My Books` page is the personalized page a patron goes to in order to access their loans, reading log, lists, and a feed of community updates:

<img width="717" alt="Screenshot 2024-12-31 at 8 21 15 AM" src="https://github.com/user-attachments/assets/c869496d-52fe-4d4b-b988-2ee5f4c88ec6" />

In #8607, a Follow feature was added, as well as a "feed" page (e.g. https://openlibrary.org/people/mekBot/books/feed) for viewing community updates

<img width="814" alt="Screenshot 2024-12-31 at 9 58 28 AM" src="https://github.com/user-attachments/assets/a2722124-00a5-4f37-8a82-a3cb46c81a45" />

Goodreads uses the homepage for similar purposes.

## Opportunity

This epic calls for the enhancement of the My Books page so:
1. it is no longer an empty, unactionable ghost town for new patrons
2. active patrons are shown fresh, relevant, actionable results that evolve over time, rather than static results
3. we can promote following as a discovery mechanism for books

If done correctly, the My Books page will become a dynamic, useful page that behaves like a monthly email news letter, showcasing what's happening in the community, what's trending, activity of those you follow, recommendations of who to follow, etc.

## Proposal 

By improving the Trending Now page #10241 to showcase patrons and the capability to follow **and** by using this component as the basis for a new section on the main My Books page, the page can be made useful to **all** registered patrons, by showcasing:
- trending books and patrons to follow (to those not yet following)
- recent activity of followed readers (for patrons already following other readers)

## Instructions

**Before asking to be assigned**, please help by:
1. Proposing a 🎨 design for Community Updates section to be added to the My Books page or...
2. By asking any clarifying questions you may have about this feature

Thank you! ❤️  

## Problem

> What problem does this proposal address & for what audience(s)?

When patrons first create an account, they are redirected to an empty My Books experience feels like a ghost town:

<img width="721" alt="Screenshot 2024-12-31 at 8 23 36 AM" src="https://github.com/user-attachments/assets/a1b76081-7e67-451d-a4b5-1dcd0b3c03ef" />

Furthermore, even active patrons (i.e. those who currently use Open Library often), are presented with a fairly static experience that is limited to showcasing actions they've taken on the website. The few/minority of patrons who follow other readers (since the feature doesn't have broad adoption yet) have to click on their `feed` link in order to see any updates from the community. 
1. This feed is unknown to many.
2. Many other patrons don't yet follow anyone because the follow function is not easily discoverable.

TL;DR:
* The follow feature is not easily discoverable
* The value of the `feed` is buried for those who do the follow feature 
* In all cases, the My Books main page is fairly static and can be evolved to solve these other two shortcomings.

## Impact

> What's the predicted impact, how do we measure, & what does success look like?

This ghost town experiences impacts every new patron. Furthermore, the experience misses out of the opportunity to better utilize the the Follow Feature #8607 -- both to showcase the rich, interesting activity of one's social network and also to recommend people to follow. 

## Research
> Are there any precedents or findings from other projects or studies that support this feature? (link or screenshot if possible)

The Goodreads homepage and activity feed is likely the best exemplar of this feature

<img width="474" alt="Screenshot 2024-12-31 at 8 33 12 AM" src="https://github.com/user-attachments/assets/cb9ed6c3-c718-4d0a-a3a0-7ad07d1f94bb" />

#### Requirements Checklist

* [ ] **Prerequisite:** #10241
* [ ] the My Books /feed styling (of the reader) should _also_ be upgraded to use similar styles and markup to the new styles we introduced in #10241 to the trending now page
* [ ] Add the new "Community Updates" social activity feed section to the My Books landing page

This proposal adds a new section to the `My Books` main page called "Community Updates", which displays up to 3 books and conditionally renders one of two experiences:

* [ ] If a patron _doesn't_ follow anyone yet (e.g. a newly registered patron), they will see books from https://openlibrary.org/trending/now with the updated design of #10241. Clicking on the title of this section should bring them to see the full trending now page.
* [ ] If a patron _does_ follow readers, this section should the last 3 items from their feed. The title of this section should be a link to see their full feed.

Other than the title and where it links to, the design/experience for this new component should be broadly the same for both experiences and look something like the trending now view and feed view:

<img width="563" alt="Screenshot 2024-12-31 at 10 03 10 AM" src="https://github.com/user-attachments/assets/57b4ee67-b28b-4e1e-926d-eb52b36fb10a" />

#### Related files

* See #6014: https://github.com/internetarchive/openlibrary/pull/6014/files#diff-b352027e4688f5ef14e5e23f6a988bba495e769f8cecce80c404a23d82f5902a

* See My Books Feed designs (e.g. https://openlibrary.org/people/mekBot/books/feed) Implemented by #8607 in
https://github.com/internetarchive/openlibrary/pull/8607/files#diff-1c0a702d7b1bad810adce7c001416b5996968d117077f629ff52ddff4668f56fR45-R51

**Comment by Spaarsh** [comment]
@mekarpeles it is indeed a huge step towards making the platform more lively for new patrons. 

I was wondering if we could ask a patron at the time of signing up to pick any 3 subjects that they would be interested in. Since the trending books only reflect the general readers and may not fit well to the needs of readers with interests that do not fall in the average set. This would ensure that the new patrons get a more relevant feed. Spotify and Medium do this for new users. 

A "Trending in {Subject}" feed is what I am thinking of. We could use the trending API and filter the results by the patron's subject of interest. If there is no book trending in that Subject, then we simply do not display its section until there is one.

The philosophy behind this is to make a new patron engage with the new environment via means of familiarity (through interested Subjects), making the initial action of reading their first book on the platform easier.

Thanks!

**Comment by mekarpeles** [comment]
@Spaarsh I think having topic selection be a new step to onboarding is a good idea we were just discussing during our last staff call, though I think we should create a separate issue to track that idea :)

**Comment by NishantSinghhhhh** [comment]
@mekarpeles can I work on this issue ??

**Comment by mekarpeles** [comment]
Hi @NishantSinghhhhh, I'm also tagging @ragipidavid & @krishnaGauss who have been collaborating on the Follow Feature and Activity Feed.

I think the piece we're currently working on is the design.

Something that would be useful is a proof-of-concept PR that fetches the right data and pulls it into the My Books summary page (i.e. https://openlibrary.org/account/books e.g. https://openlibrary.org/people/mekBot/books)

The idea is, to fetch a certain number of results (e.g. 1 page) from the feed endpoint: https://openlibrary.org/people/mekBot/books/feed and render it as a carousel under a placeholder header like "Activity Feed".

If you're willing to give that a try, I think it would be a useful/meaningful contribution.

**Comment by NishantSinghhhhh** [comment]
@mekarpeles , I am willing to take up the issue and will start working on it

**Comment by NishantSinghhhhh** [comment]
@mekarpeles , I am working on the issue, I will raise a PR once I have the code ready and all the test cases pass

**Comment by jimchamp** [comment]
Given the time constraints of this feature, and the challenges of testing following in local environments, this likely requires staff.

**Comment by mekarpeles** [comment]
@Sadashii

**Comment by Sadashii** [comment]
Interested

**Comment by mekarpeles** [comment]
Updated design elements
<img width="1024" height="209" alt="Image" src="https://github.com/user-attachments/assets/131260f8-0fa4-4de6-a838-cf17732dad35" />

1. Remove "Follow List" for list card (and fix book covers to show 3 overlapping covers side by side horizontally
2. Follow button should use our blue style shared earlier
3. The avatar, follow, and action should all be *before* the horizontal separator


Compare to goodreads design
<img width="300"  alt="Image" src="https://github.com/user-attachments/assets/ef922381-96df-4926-909e-b361d16f20c5" />

**Comment by mekarpeles** [comment]
How do we feel about all patrons following the Open Library Community Account by default? Or does it make more sense to show activity across the entire library (until you follow your first person)?

**Comment by mekarpeles** [comment]
Possible Activity Feed design with...

Discover | Following | Popular (which remembers your selection or defaults to Following if not empty)

<img width="910" height="952" alt="Image" src="https://github.com/user-attachments/assets/219a2074-2e36-4ed5-bd82-23fbd22ddf19" />
